src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/TargetFrameworkProvider.cs

```diff
@@ -76,22 +38,30 @@ public ITargetFramework GetTargetFramework(string shortOrFullName)
             ITargetFramework targetFramework = null;
             try
             {
+                // Fast path assuming that we've encountered this name before.
+                if (CachedTargetFrameworks.TryGetValue(shortOrFullName, out targetFramework))
```

Comment:
Dictionary is not thread safe for a reader and writer at the same time, I think you'd better off using a ConcurrentDictionary.